components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/model/ServiceProvider.java

```diff
@@ -320,4 +323,41 @@ public void setSaasApp(boolean saasApp) {
     public void setSpProperties(ServiceProviderProperty[] spProperties) {
         this.spProperties = spProperties;
     }
+
+    /**
+     *
+     * Returns the certificate content
+     *
+     * @return the certificate in PEM format.
+     */
+    public String getCertificateContent() {
+        return certificateContent;
+    }
+
+    /**
+     *
+     * Sets the given certificate.
+     *
+     * @param certificateContent the certificate in PEM format
+     */
+    public void setCertificateContent(String certificateContent) {
+        this.certificateContent = certificateContent;
+    }
+
+    /**
+     * Sets whether the per SP certificate support is available.
+     * TODO: Get rid of this in the public branch since this was introduced to safeguard the backward compatibility.
```

Comment:
Since this is the public branch do we need this?

Reply:
Rather than having a separate property introduced in the service provider to ensure backward compatibility I think when certificate is being picked for validations and encryption, if SP specific cert is not to be found we can switch back to the previous mechanism used to pick up certificate.

Reply:
@malithie This property was added to conditionally display the 'Application Certificate' text area when the database support is available. Though this approach looks ugly, I couldn't think of another way since we can't depend on the Utils methods of other components.

One other solution is to do the same in the UI OSGi component class as [this](https://github.com/rushmin/carbon-identity-framework/blob/90fb0666a4fbb089038d223bd573f62a7a1cb8d7/components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/internal/ApplicationManagementServiceComponent.java#L240). But the downside is duplicate code. Any thoughts?

@mefarazath, We are not going to remove the backward compatibility since this is the first public release which contains this feature. So we will be deprecating the previous implementation in this release and remove it in a later version. I will update the comment accordingly.